### PR TITLE
Fixings fstrings in pandas/io/stata.py & pandas/io/sql.py

### PR DESCRIPTION
**Pull request by CortlandMorse**
- [x] ref #29547
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/sql.py [hunk 1]
```diff
@@ -1599,14 +1588,11 @@ def execute(self, *args, **kwargs):
                 self.con.rollback()
             except Exception as inner_exc:  # pragma: no cover
                 ex = DatabaseError(
-                    "Execution failed on sql: {sql}\n{exc}\nunable "
-                    "to rollback".format(sql=args[0], exc=exc)
+                    f"Execution failed on sql: {args[0]}\n{exc}\nunable " "to rollback"
```
Comment (WillAyd) [inline pandas/io/sql.py]:
```suggestion
                    f"Execution failed on sql: {args[0]}\n{exc}\nunable to rollback"
```

File: pandas/io/stata.py [hunk 2]
```diff
@@ -590,20 +569,18 @@ def _cast_to_stata_types(data):
...
                 data[col] = data[col].astype(np.float64)
             elif dtype == np.float64:
                 if value > float64_max:
                     raise ValueError(
-                        "Column {col} has a maximum value "
-                        "({val}) outside the range supported by "
-                        "Stata ({float64_max})".format(
-                            col=col, val=value, float64_max=float64_max
-                        )
+                        f"Column {col} has a maximum value "
```
Comment (WillAyd) [inline pandas/io/stata.py]: I think some of this might conflict with #30285

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
Thanks for the PR! CI failure is unrelated a few comments on first glance

**Comment by bashtage** [comment]
This is going to produce a lot of conflicts with #30285

File: pandas/io/stata.py [hunk 3]
```diff
@@ -497,29 +495,10 @@ class ValueLabelTypeMismatch(Warning):
     pass
 
 
-value_label_mismatch_doc = """
-Stata value labels (pandas categories) must be strings. Column {0} contains
-non-string labels which will be converted to strings.  Please check that the
-Stata data file created has not lost information due to duplicate labels.
-"""
-
-
```
Comment (bashtage) [inline pandas/io/stata.py]: I'm not a big fan of moving this very long warning to be inline just to make it an f-string.  f-strings should not be dogmatically used when they produce worse code.

File: pandas/io/stata.py [hunk 4]
```diff
@@ -497,29 +495,10 @@ class ValueLabelTypeMismatch(Warning):
...
-invalid_name_doc = """
-Not all pandas column names were valid Stata variable names.
-The following replacements have been made:
-
-    {0}
-
-If this is not what you expect, please make sure you have Stata-compliant
-column names in your DataFrame (strings only, max 32 characters, only
-alphanumerics and underscores, no Stata reserved words)
-"""
```
Comment (bashtage) [inline pandas/io/stata.py]: Same comment here.

**Review by bashtage — COMMENTED** [review]
I think some of the changes are not improvements.

File: pandas/io/sql.py [hunk 5]
```diff
@@ -1304,10 +1300,7 @@ def to_sql(
 
             for col, my_type in dtype.items():
                 if not isinstance(to_instance(my_type), TypeEngine):
-                    raise ValueError(
-                        "The type of {column} is not a "
-                        "SQLAlchemy type ".format(column=col)
-                    )
+                    raise ValueError(f"The type of {col} is not a " "SQLAlchemy type ")
```
Comment (jbrockmendel) [inline pandas/io/sql.py]: looks like there is an extra space after "type", and black formatting split this into two strings that should be manually re-joined.

File: pandas/io/stata.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/io/stata.py]: agreed

File: pandas/io/stata.py (hunk 4 above)
Comment (jbrockmendel) [reply pandas/io/stata.py]: Agreed again.  Though if you want to change `{0}` to use a named argument, id be on board with that

File: pandas/io/stata.py [hunk 6]
```diff
@@ -1330,19 +1313,13 @@ def _read_old_header(self, first_char):
         try:
             self.typlist = [self.TYPE_MAP[typ] for typ in typlist]
         except ValueError:
-            raise ValueError(
-                "cannot convert stata types [{0}]".format(
-                    ",".join(str(x) for x in typlist)
-                )
-            )
+            errjoin = ", ".join(str(x) for x in typlist)
+            raise ValueError("cannot convert stata types " f"[{errjoin}]")
```
Comment (jbrockmendel) [inline pandas/io/stata.py]: need to un-split string

File: pandas/io/stata.py [hunk 7]
```diff
@@ -1330,19 +1313,13 @@ def _read_old_header(self, first_char):
...
         try:
             self.dtyplist = [self.DTYPE_MAP[typ] for typ in typlist]
         except ValueError:
-            raise ValueError(
-                "cannot convert stata dtypes [{0}]".format(
-                    ",".join(str(x) for x in typlist)
-                )
-            )
+            errjoin = ", ".join(str(x) for x in typlist)
+            raise ValueError("cannot convert stata dtypes " f"[{errjoin}]")
```
Comment (jbrockmendel) [inline pandas/io/stata.py]: need to unsplit string

File: pandas/io/stata.py (hunk 3 above)
Comment (CortlandMorse) [reply pandas/io/stata.py]: makes sense, I'll change it in my next commit

File: pandas/io/sql.py (hunk 5 above)
Comment (CortlandMorse) [reply pandas/io/sql.py]: i'll join it in my next commit, thanks for the feedback

File: pandas/io/stata.py (hunk 6 above)
Comment (CortlandMorse) [reply pandas/io/stata.py]: oops, thanks Ill fix it

File: pandas/io/stata.py [hunk 8]
```diff
@@ -1794,18 +1773,16 @@ def _do_convert_categoricals(
...
-                    raise ValueError(msg.format(col=col, repeats=repeats))
+                    msg = (
+                        f"\nValue labels for column {col} are not unique. These "
+                        "cannot be converted to\npandas categoricals.\n\n"
+                        "Either read the file with `convert_categoricals` set "
+                        "to False or use the\nlow level interface in "
+                        "`StataReader` to separately read the values and the"
+                        "\nvalue_labels.\n\nThe repeated labels are:\n"
+                        f"{repeats}\n"
+                    )
```
Comment (jbrockmendel) [inline pandas/io/stata.py]: Some of these newlines can be removed.  Only keep the ones that separate paragraphs

Comment (jbrockmendel) [reply pandas/io/stata.py]: same above

**Comment by datapythonista** [comment]
While the changes here look cool, I think we should close this PR, and leave the changes here to #30285 (and a follow up of if needed).

**Comment by jbrockmendel** [comment]
agreed with @datapythonista.  Will revisit after #30285.  Closing.

**Pull request closed without merging**